core/trino-main/src/main/java/io/trino/ExceededMemoryLimitException.java

```diff
@@ -40,6 +42,12 @@ public static ExceededMemoryLimitException exceededLocalUserMemoryLimit(DataSize
                 format("Query exceeded per-node memory limit of %s [%s]", maxMemory, additionalFailureInfo));
     }
 
+    public static ExceededMemoryLimitException exceededConnectorMemoryLimit(ConnectorName connectorName, DataSize maxMemory, String additionalFailureInfo)
```

Comment:
This is only used in one place. Inline it.

Reply:
Do you mean changing the error to public and throwing it directly from CatalogConnector?

Reply:
Also, take a look at [#22197 (comment)](https://github.com/trinodb/trino/pull/22197#:~:text=Look%20at-,%2322197%20(comment),-.%20I%20think%20we)

Reply:
From our offline conversation, let's keep this for now, as it's how it's being used in other places. In the future, we need to clean up this class by making the constructor public and letting callers decide what error codes and messages to use. It's not a good pattern to have the exception class know about all possible usages.